include/MFEMMGIS/ParaviewExportResults.ixx

```diff
@@ -8,47 +8,177 @@
-      NonLinearEvolutionProblemImplementation<parallel>& p,
-      const Parameters& params)
-      : exporter(get<std::string>(params, "OutputFileName"),
-                 p.getFiniteElementSpace().GetMesh()),
-        result(&p.getFiniteElementSpace()),
-        cycle(0) {
-    auto& u1 = p.getUnknownsAtEndOfTheTimeStep();
-    this->result.MakeTRef(&p.getFiniteElementSpace(), u1, 0);
-    if (contains(params, "OutputFieldName")) {
-      this->exporter.RegisterField(get<std::string>(params, "OutputFieldName"),
-                                   &(this->result));
-    } else {
-      this->exporter.RegisterField("u", &(this->result));
+  template<typename Mesh>
+    void print_mesh_information(Mesh* mesh)
+    {
+
+      using Profiler::Utils::sum;
+      using Profiler::Utils::Message;
+
+      //get the number of vertices
+      int64_t numbers_of_vertices_local = mesh->GetNV();
+      int64_t  numbers_of_vertices = sum(numbers_of_vertices_local);
+
+      //get the number of elements
+      int64_t numbers_of_elements_local = mesh->GetNE();
+      int64_t numbers_of_elements = sum(numbers_of_elements_local);
+
+      Message("INFO: number of vertices -> ", numbers_of_vertices);
+      Message("INFO: number of elements -> ", numbers_of_elements);
     }
-  }  // end of ParaviewExportResults
 
   template <bool parallel>
-  void ParaviewExportResults<parallel>::execute(
-      NonLinearEvolutionProblemImplementation<parallel>&,
-      const real t,
-      const real dt) {
-    this->exporter.SetCycle(this->cycle);
-    this->exporter.SetTime(t + dt);
-    // SetFromTrueVector needed here in MFEM for at least two rationales:
-    //    - it applies prolongation matrix (Non-Conforming mesh, BCs, AMR ...)
-    //      to set the values of some unkwown dofs deduced from known dofs
-    //    - exchange data between processes in order to retrieve information
-    //      needed to perform the previous prolongation step
-    this->result.SetFromTrueVector();
-    this->exporter.Save();
-    ++(this->cycle);
-  }  // end of execute
+    ParaviewExportResults<parallel>::ParaviewExportResults(
+        NonLinearEvolutionProblemImplementation<parallel>& p,
+        const Parameters& params)
+    : exporter(get<std::string>(params, "OutputFileName")),
+    result(&p.getFiniteElementSpace()),
+    cycle(0) {
+      CatchTimeSection("ParaviewExportResults::Constructor");
+      auto& u1 = p.getUnknownsAtEndOfTheTimeStep();
+      this->result.MakeTRef(&p.getFiniteElementSpace(), u1, 0);
+
+      /** Default, the mesh is the entire mesh */
+      Mesh<parallel>& pmesh = p.getMesh();
+
+      if(contains(params, "DomainAttributes") && contains(params, "BoundaryAttributes"))
+      {
+        Profiler::Utils::Message("You can not define 'DomainAttributes' and 'BoundaryAttributes' in a single ParaviewExportResults post processing");
+        std::exit(0);
+      }
+
+      if(contains(params, "DomainAttributes")){ /** DomainAttributes and Sub mesh */
```

Comment:
The rest of `MFEM/MGIS` allows to specify an indentifier or a name, please be consistent with this.

See `FiniteElementDiscretization::getMaterialsIdentifiers` and  `FiniteElementDiscretization::getBoundariesIdentifiers` for details